### PR TITLE
Create a new 25.1.0 release

### PR DESCRIPTION
**Pull request by piyush-nimbalkar**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,7 +28,7 @@ ifndef DOCKER_HUB_REGISTRY_IMG
     $(warning DOCKER_HUB_REGISTRY_IMG not defined, using '$(DOCKER_HUB_REGISTRY_IMG)' instead)
 endif
 ifndef BASE_REGISTRY_IMG
-    BASE_REGISTRY_IMG := docker.io/portworx/px-operator-registry:23.10.3
+    BASE_REGISTRY_IMG := docker.io/portworx/px-operator-registry:25.1.0
```
Comment (piyush-nimbalkar) [inline Makefile:31]: This should be the previous release version.

File: deploy/olm-catalog/portworx/25.1.0/portworx-certified.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+          },
+          "spec": {}
+        }
+      ]
+spec:
+  displayName: Portworx Enterprise
+  version: 25.1.0
+  minKubeVersion: "1.21.0"
+  maturity: stable
+  replaces: portworx-operator.v25.1.0
```
Comment (piyush-nimbalkar) [inline deploy/olm-catalog/portworx/25.1.0/portworx-certified.clusterserviceversion.yaml:56]: This should be the previous release version.

File: Makefile
Comment (piyush-nimbalkar) [inline Makefile:1]: Line number 53 also needs to change to 
RELEASE_VER := 25.1.0

File: Makefile (hunk 1 above)
Comment (mborodin-px) [reply Makefile]: fixed

File: Makefile
Comment (mborodin-px) [reply Makefile:1]: it should not, we have to change it to 99.9.9 to avoid if someone will run release target

**Pull request closed without merging**